### PR TITLE
Commerce 1453 (SF fixes from Luca)

### PR DESCRIPTION
**Pull request by alecsloan**
From #1172

### CODE REVIEW

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

**Comment by ethanbustad** [comment]
@alecsloan requires another rebase :(

**Comment by alecsloan** [comment]
I'll rebase when @marco-leo's ready to merge ;)

File: commerce-catalog-web/src/main/java/com/liferay/commerce/catalog/web/internal/display/context/BaseCommerceCatalogDisplayContext.java [hunk 1]
```diff
@@ -58,10 +58,10 @@ public PortletURL getPortletURL() throws PortalException {
 		return portletURL;
 	}
 
-	protected final CPRequestHelper cpRequestHelper;
 	protected final HttpServletRequest httpServletRequest;
 	protected final LiferayPortletRequest liferayPortletRequest;
 	protected final LiferayPortletResponse liferayPortletResponse;
 	protected final PortalPreferences portalPreferences;
+	protected final CommerceCatalogRequestHelper requestHelper;
```
Comment (ethanbustad) [inline commerce-catalog-web/src/main/java/com/liferay/commerce/catalog/web/internal/display/context/BaseCommerceCatalogDisplayContext.java:65]: `CommerceCatalogRequestHelper requestHelper` should be `CommerceCatalogRequestHelper commerceCatalogRequestHelper`

File: commerce-channel-web/src/main/java/com/liferay/commerce/channel/web/internal/display/context/BaseCommerceChannelDisplayContext.java [hunk 2]
```diff
@@ -58,10 +58,10 @@ public PortletURL getPortletURL() throws PortalException {
 		return portletURL;
 	}
 
-	protected final CPRequestHelper cpRequestHelper;
 	protected final HttpServletRequest httpServletRequest;
 	protected final LiferayPortletRequest liferayPortletRequest;
 	protected final LiferayPortletResponse liferayPortletResponse;
 	protected final PortalPreferences portalPreferences;
+	protected final CommerceChannelRequestHelper requestHelper;
```
Comment (ethanbustad) [inline commerce-channel-web/src/main/java/com/liferay/commerce/channel/web/internal/display/context/BaseCommerceChannelDisplayContext.java:65]: `CommerceCatalogRequestHelper requestHelper` should be `CommerceCatalogRequestHelper commerceCatalogRequestHelper`

File: commerce-item-selector-web/src/main/java/com/liferay/commerce/item/selector/web/internal/display/context/BaseCommerceItemSelectorViewDisplayContext.java [hunk 3]
```diff
@@ -79,14 +80,14 @@ protected String getKeywords() {
...
-			cpRequestHelper.getRenderRequest(), "keywords");
+			requestHelper.getRenderRequest(), "keywords");
 
 		return _keywords;
 	}
 
-	protected final CPRequestHelper cpRequestHelper;
 	protected final HttpServletRequest httpServletRequest;
 	protected final String itemSelectedEventName;
+	protected final CommerceItemSelectorRequestHelper requestHelper;
```
Comment (ethanbustad) [inline commerce-item-selector-web/src/main/java/com/liferay/commerce/item/selector/web/internal/display/context/BaseCommerceItemSelectorViewDisplayContext.java:90]: `CommerceCatalogRequestHelper requestHelper` should be `CommerceCatalogRequestHelper commerceCatalogRequestHelper`

File: commerce-order-content-web/src/main/java/com/liferay/commerce/order/content/web/internal/display/context/CommerceOrderContentDisplayContext.java [hunk 4]
```diff
@@ -493,7 +493,7 @@ public boolean isShowPurchaseOrderNumber() throws PortalException {
 	private final CommercePaymentMethodGroupRelService
 		_commercePaymentMethodGroupRelService;
 	private final CommerceShipmentItemService _commerceShipmentItemService;
-	private final CPRequestHelper _cpRequestHelper;
+	private final CommerceOrderRequestHelper _cpRequestHelper;
```
Comment (ethanbustad) [inline commerce-order-content-web/src/main/java/com/liferay/commerce/order/content/web/internal/display/context/CommerceOrderContentDisplayContext.java:496]: `CommerceOrderRequestHelper _cpRequestHelper` should be `CommerceOrderRequestHelper _commerceOrderRequestHelper`

File: commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/display/context/CPSearchResultsDisplayContext.java [hunk 5]
```diff
@@ -318,7 +318,7 @@ protected String getURLString() {
 		_cpContentListEntryRendererRegistry;
 	private final CPContentListRendererRegistry _cpContentListRendererRegistry;
 	private final CPDefinitionHelper _cpDefinitionHelper;
-	private final CPRequestHelper _cpRequestHelper;
+	private final CPSearchResultsRequestHelper _cpRequestHelper;
```
Comment (ethanbustad) [inline commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/display/context/CPSearchResultsDisplayContext.java:321]: `CPSearchResultsRequestHelper _cpRequestHelper` should be `CPSearchResultsRequestHelper _cpSearchResultsRequestHelper`

File: commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/display/context/CPCompareContentDisplayContext.java [hunk 6]
```diff
@@ -244,7 +244,7 @@ public boolean isSelectionStyleCustomRenderer() {
 	private final CPContentListRendererRegistry _cpContentListRendererRegistry;
 	private final CPDefinitionHelper _cpDefinitionHelper;
 	private final List<Long> _cpDefinitionIds;
-	private final CPRequestHelper _cpRequestHelper;
+	private final CPContentRequestHelper _cpRequestHelper;
```
Comment (ethanbustad) [inline commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/display/context/CPCompareContentDisplayContext.java:247]: `CPContentRequestHelper _cpRequestHelper` should be `CPContentRequestHelper _cpContentRequestHelper`

File: commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/display/context/CPCompareContentMiniDisplayContext.java [hunk 7]
```diff
@@ -247,7 +247,7 @@ public boolean isSelectionStyleCustomRenderer() {
 	private final CPContentListRendererRegistry _cpContentListRendererRegistry;
 	private final CPDefinitionHelper _cpDefinitionHelper;
 	private final List<Long> _cpDefinitionIds;
-	private final CPRequestHelper _cpRequestHelper;
+	private final CPContentRequestHelper _cpRequestHelper;
```
Comment (ethanbustad) [inline commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/display/context/CPCompareContentMiniDisplayContext.java:250]: `CPContentRequestHelper _cpRequestHelper` should be `CPContentRequestHelper _cpContentRequestHelper`

File: commerce-product-item-selector-web/src/main/java/com/liferay/commerce/product/item/selector/web/internal/display/context/BaseCPItemSelectorViewDisplayContext.java [hunk 8]
```diff
@@ -194,11 +194,11 @@ protected RowChecker getRowChecker() {
 		return _rowChecker;
 	}
 
-	protected final CPRequestHelper cpRequestHelper;
 	protected final HttpServletRequest httpServletRequest;
 	protected final LiferayPortletRequest liferayPortletRequest;
 	protected final LiferayPortletResponse liferayPortletResponse;
 	protected final PortalPreferences portalPreferences;
+	protected final CPItemSelectorRequestHelper requestHelper;
```
Comment (ethanbustad) [inline commerce-product-item-selector-web/src/main/java/com/liferay/commerce/product/item/selector/web/internal/display/context/BaseCPItemSelectorViewDisplayContext.java:201]: `CPItemSelectorRequestHelper requestHelper` should be `CPItemSelectorRequestHelper cpItemSelectorRequestHelper`

File: commerce-product-options-web/src/main/java/com/liferay/commerce/product/options/web/internal/display/context/BaseCPOptionsDisplayContext.java [hunk 9]
```diff
@@ -230,11 +230,11 @@ protected void setDefaultOrderByType(String defaultOrderByType) {
 	}
 
 	protected final ActionHelper actionHelper;
-	protected final CPRequestHelper cpRequestHelper;
 	protected final HttpServletRequest httpServletRequest;
 	protected final LiferayPortletRequest liferayPortletRequest;
 	protected final LiferayPortletResponse liferayPortletResponse;
 	protected final PortalPreferences portalPreferences;
+	protected final CPOptionsRequestHelper requestHelper;
```
Comment (ethanbustad) [inline commerce-product-options-web/src/main/java/com/liferay/commerce/product/options/web/internal/display/context/BaseCPOptionsDisplayContext.java:237]: `CPOptionsRequestHelper requestHelper` should be `CPOptionsRequestHelper cpOptionsRequestHelper`

File: commerce-shipment-web/src/main/java/com/liferay/commerce/shipment/web/internal/display/context/BaseCommerceShipmentDisplayContext.java [hunk 10]
```diff
@@ -257,11 +257,11 @@ protected String getDisplayStyle(
 	}
 
 	protected final ActionHelper actionHelper;
-	protected final CPRequestHelper cpRequestHelper;
 	protected final HttpServletRequest httpServletRequest;
 	protected final LiferayPortletRequest liferayPortletRequest;
 	protected final LiferayPortletResponse liferayPortletResponse;
 	protected final PortalPreferences portalPreferences;
+	protected final CommerceShipmentRequestHelper requestHelper;
```
Comment (ethanbustad) [inline commerce-shipment-web/src/main/java/com/liferay/commerce/shipment/web/internal/display/context/BaseCommerceShipmentDisplayContext.java:264]: `CommerceShipmentRequestHelper requestHelper` should be `CommerceShipmentRequestHelper commerceShipmentRequestHelper`

File: commerce-subscription-web/src/main/java/com/liferay/commerce/subscription/web/internal/display/context/CommerceSubscriptionContentDisplayContext.java [hunk 11]
```diff
@@ -179,7 +180,7 @@ public PortletURL getPortletURL() throws PortalException {
 	private final ConfigurationProvider _configurationProvider;
 	private final CPDefinitionHelper _cpDefinitionHelper;
 	private final CPInstanceHelper _cpInstanceHelper;
-	private final CPRequestHelper _cpRequestHelper;
+	private final CommerceSubscriptionRequestHelper _cpRequestHelper;
```
Comment (ethanbustad) [inline commerce-subscription-web/src/main/java/com/liferay/commerce/subscription/web/internal/display/context/CommerceSubscriptionContentDisplayContext.java:189]: `CommerceSubscriptionRequestHelper _cpRequestHelper` should be `CommerceSubscriptionRequestHelper _commerceSubscriptionRequestHelper`

File: commerce-subscription-web/src/main/java/com/liferay/commerce/subscription/web/internal/display/context/CommerceSubscriptionEntryDisplayContext.java [hunk 12]
```diff
@@ -417,7 +418,7 @@ protected String getNavigation() {
 	private final CommerceSubscriptionEntryService
 		_commerceSubscriptionEntryService;
 	private final ConfigurationProvider _configurationProvider;
-	private final CPRequestHelper _cpRequestHelper;
+	private final CommerceSubscriptionRequestHelper _cpRequestHelper;
```
Comment (ethanbustad) [inline commerce-subscription-web/src/main/java/com/liferay/commerce/subscription/web/internal/display/context/CommerceSubscriptionEntryDisplayContext.java:421]: `CommerceSubscriptionRequestHelper _cpRequestHelper` should be `CommerceSubscriptionRequestHelper _commerceSubscriptionRequestHelper`

File: commerce-price-list-web/src/main/java/com/liferay/commerce/price/list/web/internal/display/context/BaseCommercePriceListDisplayContext.java [hunk 13]
```diff
@@ -210,6 +210,7 @@ protected String getNavigation() {
 	protected final HttpServletRequest httpServletRequest;
 	protected final LiferayPortletRequest liferayPortletRequest;
 	protected final LiferayPortletResponse liferayPortletResponse;
+	protected CommercePriceListRequestHelper requestHelper;
```
Comment (ethanbustad) [inline commerce-price-list-web/src/main/java/com/liferay/commerce/price/list/web/internal/display/context/BaseCommercePriceListDisplayContext.java:213]: `CommercePriceListRequestHelper requestHelper` should be `CommercePriceListRequestHelper commercePriceListRequestHelper`

File: commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/portlet/action/CPSearchResultsConfigurationAction.java [hunk 14]
```diff
@@ -48,8 +48,8 @@
 
 	@Override
 	public String getJspPath(HttpServletRequest httpServletRequest) {
-		CPRequestHelper cpRequestHelper = new CPRequestHelper(
-			httpServletRequest);
+		CPSearchResultsRequestHelper cpRequestHelper =
```
Comment (ethanbustad) [inline commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/portlet/action/CPSearchResultsConfigurationAction.java:51]: `CPSearchResultsRequestHelper cpRequestHelper` should be `CPSearchResultsRequestHelper cpSearchResultsRequestHelper`

File: commerce-product-definitions-api/src/main/java/com/liferay/commerce/product/definitions/display/context/BaseCPDefinitionsDisplayContext.java [hunk 15]
```diff
@@ -199,11 +200,11 @@ public boolean hasPermission(
 	}
 
 	protected final ActionHelper actionHelper;
-	protected final CPRequestHelper cpRequestHelper;
 	protected final HttpServletRequest httpServletRequest;
 	protected final LiferayPortletRequest liferayPortletRequest;
 	protected final LiferayPortletResponse liferayPortletResponse;
 	protected final PortalPreferences portalPreferences;
+	protected final CPDefinitionRequestHelper requestHelper;
```
Comment (ethanbustad) [inline commerce-product-definitions-api/src/main/java/com/liferay/commerce/product/definitions/display/context/BaseCPDefinitionsDisplayContext.java:207]: same

File: commerce-product-definitions-api/src/main/java/com/liferay/commerce/product/definitions/display/context/BaseCPDefinitionsSearchContainerDisplayContext.java [hunk 16]
```diff
@@ -0,0 +1,301 @@
...
+		managementBarFilterItems.add(
+			getManagementBarFilterItem(WorkflowConstants.STATUS_ANY));
+		managementBarFilterItems.add(
+			getManagementBarFilterItem(WorkflowConstants.STATUS_DRAFT));
+		managementBarFilterItems.add(
+			getManagementBarFilterItem(WorkflowConstants.STATUS_SCHEDULED));
+		managementBarFilterItems.add(
+			getManagementBarFilterItem(WorkflowConstants.STATUS_APPROVED));
+		managementBarFilterItems.add(
+			getManagementBarFilterItem(WorkflowConstants.STATUS_EXPIRED));
```
Comment (ethanbustad) [inline commerce-product-definitions-api/src/main/java/com/liferay/commerce/product/definitions/display/context/BaseCPDefinitionsSearchContainerDisplayContext.java]: If order matters here, we should add blank lines in between each `add` to make that clear

**Review by ethanbustad — CHANGES_REQUESTED** [review]

**Comment by ethanbustad** [comment]
Oops, @luca-pellizzon @marco-leo, we had some duplicated efforts. My bad. I had already removed the `commerce-product-service` exports under a different ticket in this PR: https://github.com/liferay/com-liferay-commerce/pull/1486. I like some of these changes better than the ones I made, so @marco-leo feel free to just merge this one (or take a look and see what solutions you like better -- we could always do a mix of the 2, too).

**Comment by marco-leo** [comment]
Sorry, too risky to merge this pull, too many changed introduces, difficult to track

**Comment by marco-leo** [comment]
@ethanbustad @luca-pellizzon  if we want those changes in we should send smaller PRs. At this point pull like these can't be merged

**Pull request closed without merging**